src/nvim/api/autocmd.c

```diff
@@ -57,15 +57,26 @@ Array nvim_get_autocmds(Dict(get_autocmds) *opts, Error *err)
 
   if (opts->group.type != kObjectTypeNil) {
     Object v = opts->group;
-    if (v.type != kObjectTypeString) {
-      api_set_error(err, kErrorTypeValidation, "group must be a string.");
-      goto cleanup;
-    }
+    switch (v.type) {
+    case kObjectTypeString:
```

Comment:
`kObjectTypeNil` is just another type, so you can include it in the switch statement:

```c
switch (opts->group.type) {
case kObjectTypeString:
  // handle string case
  break;
case kObjectTypeInteger:
  // handle int case
  break;
case kObjectTypeNil:
  // do nothing
  break;
default:
  // error
  api_set_error(...);
  break;
}
```